File: Stack/Min_Stack/1_BruteForce.cpp
#include <stack>
#include <climits>
using namespace std;

class MinStack {
    stack<int> st;
public:
    MinStack() {}
    void push(int x) { st.push(x); }
    void pop() { if(!st.empty()) st.pop(); }
    int top() { return st.top(); }
    int getMin() {
        int mn = INT_MAX;
        stack<int> temp = st;
        while(!temp.empty()) {
            mn = min(mn, temp.top());
            temp.pop();
        }
        return mn;
    }
};

File: Stack/Min_Stack/1_BruteForce.java
import java.util.Stack;

public class MinStack {
    private Stack<Integer> stack;
    public MinStack() {
        stack = new Stack<>();
    }
    public void push(int x) {
        stack.push(x);
    }
    public void pop() {
        if(!stack.isEmpty()) stack.pop();
    }
    public int top() {
        return stack.peek();
    }
    public int getMin() {
        int min = Integer.MAX_VALUE;
        for(int val : stack) {
            if(val < min) min = val;
        }
        return min;
    }
}
